ddl/ddl_api.go

```diff
@@ -5741,7 +5741,7 @@ func isTableTiFlashSupported(schema *model.DBInfo, tb table.Table) error {
 
 func checkTiFlashReplicaCount(ctx sessionctx.Context, replicaCount uint64) error {
 	// Check the tiflash replica count should be less than the total tiflash stores.
-	tiflashStoreCnt, err := infoschema.GetTiFlashWriteStoreCount(ctx)
+	tiflashStoreCnt, err := infoschema.GetTiFlashStoreCount(ctx)
```

Comment:
Actually I believe we still need to only count write nodes here. But we could fix it (along with elsewhere that requires write nodes) later.

Reply:
> Actually I believe we still need to only count write nodes here. But we could fix it (along with elsewhere that requires write nodes) later.

We don't modify `serverType` now, the `serverType` of `tiflash_write` is `tiflash`, so `tiflashStoreCnt` includes the count of `tiflash_write` and non-disaggregated TiFlash.